### ISSUE TITLE
Run multiple docker image on Travis

### ISSUE THREAD

**Issue by junghans** [losalamos/flecsi#72]
@ollielo: It seems like we can run docker inside travis:
https://docs.travis-ci.com/user/docker/

**Comment by ollielo** [comment]
@junghans 
The docker package in the Travis CI (ubuntu 14.04) is too old to support the `--build-arg` option to pass environment variable from the matrix to `docker build`.  Now we either have to "sed Dockerfile" or install newer docker package in the Travis build environment.

**Comment by junghans** [comment]
With `sudo: required` you should be able to use docker's deb repository:
https://docs.docker.com/engine/installation/linux/ubuntulinux/
e.g.
```
sudo add-apt-repository "deb https://apt.dockerproject.org/repo ubuntu-trusty main"
```

**Comment by ollielo** [comment]
I went the "sed dockerfile" way. It work for Fedora with Ubuntu on the way.

**Comment by junghans** [comment]
This is really ugly, is there a good reason to not update docker using the above repo?

**Comment by ollielo** [comment]
There is no good reason to not use updated docker package. The only reason
is that I am not familiar with apt-get based system.

**Comment by ollielo** [comment]
@junghans 

It looks like there is no way to compile with -Werror with the combination of docker support

**Comment by ollielo** [comment]
Closed with PR#79.
